File: react-native-nitro-biometrics/ios/HybridNitroBiometrics.swift
//
//  HybridNitroBiometrics.swift
//  Pods
//
//  Created by frankudoags on 5/19/2026.
//

import Foundation
import NitroModules

class HybridNitroBiometrics: HybridNitroBiometricsSpec {
    func isAvailable() throws -> BiometricsAvailability {
        
    }
    
    func authenticate(reason: String) throws -> NitroModules.Promise<BiometricsAuthResult> {
        <#code#>
    }
    
    func createKeys() throws -> NitroModules.Promise<BiometricsKey> {
        <#code#>
    }
    
    func signPayload(payload: String) throws -> NitroModules.Promise<BiometricsSignature> {
        <#code#>
    }
    
    func deleteKeys() throws {
        <#code#>
    }
    
    
}

File: react-native-nitro-biometrics/android/src/main/java/com/nitrobiometrics/HybridNitroBiometrics.kt
package com.nitrobiometrics

import com.margelo.nitro.core.Promise
import com.margelo.nitro.nitrobiometrics.BiometricsAuthResult
import com.margelo.nitro.nitrobiometrics.BiometricsAvailability
import com.margelo.nitro.nitrobiometrics.BiometricsKey
import com.margelo.nitro.nitrobiometrics.BiometricsSignature
import com.margelo.nitro.nitrobiometrics.HybridNitroBiometricsSpec

class HybridNitroBiometrics: HybridNitroBiometricsSpec() {
    override fun isAvailable(): BiometricsAvailability {
        TODO("Not yet implemented")
    }

    override fun authenticate(reason: String): Promise<BiometricsAuthResult> {
        TODO("Not yet implemented")
    }

    override fun createKeys(): Promise<BiometricsKey> {
        TODO("Not yet implemented")
    }

    override fun signPayload(payload: String): Promise<BiometricsSignature> {
        TODO("Not yet implemented")
    }

    override fun deleteKeys() {
        TODO("Not yet implemented")
    }
}
